### PR TITLE
fix for: ReferenceError: v is not defined and improved regex

### PR DESCRIPTION
**Pull request by klict**
Hello,

this is a small fix for uncaught ReferenceError: v is not defined ;

You guys were executing function test on variable v but i think you should use variable value since variable v is not defined anywhere.

Also my team leader has improved your the regex for "validate-clean-url" and "validate-url".

credit for regex @redelschaap.

### CODE REVIEW

**Comment by redelschaap** [comment]
Note for my improved regex: [RFC 1035 section 2.3.4](http://www.freesoft.org/CIE/RFC/1035/9.htm) states that each part of a domain name can contain up to 63 characters, which means also the TLD extension can technically contain 63 charachters. The regex checks whether the DNS part of the URL contains valid parts and is not restricted to the set of TLD's of the original regex.

**Comment by sitepodmatt** [comment]
Does this work with http://unicode.org/faq/idn.html?

**Comment by redelschaap** [comment]
As a matter of fact it doesn't. But I have only improved the DNS part, so neither does the original.

These regex patterns do the trick. Since it's very hard to check for all allowed charachters in the path, query and fragment parts (for example, Chinese or Arabic letters are allowed too), it checks the three parts for whitespace characters only.

@klict Can you please change the regex patterns in your commit?

**validate-url**

```
return /^((http|https|ftp):\/\/)(([A-Z0-9][A-Z0-9_-]{1,62}\.)+([A-Z]{2,63}(\.[A-Z]{2,63})?))(:(\d+))?(\/[^\?#\s]*)?(\?[^#\s]*)?(#[^\s]*)?$/i.test(value);
```

**validate-clean-url**

```
return utils.isEmptyNoTrim(value) || /^((http|https|ftp):\/\/)?(([A-Z0-9][A-Z0-9_-]{1,62}\.)+([A-Z]{2,63}(\.[A-Z]{2,63})?))(:(\d+))?(\/[^\?#\s]*)?(\?[^#\s]*)?(#[^\s]*)?$/i.test(value);
```

File: app/code/Magento/Ui/view/base/web/js/lib/validation/rules.js [hunk 1]
```diff
@@ -468,14 +468,14 @@ define([
...
-                return (/^(http|https|ftp):\/\/(([A-Z0-9]([A-Z0-9_-]*[A-Z0-9]|))(\.[A-Z0-9]([A-Z0-9_-]*[A-Z0-9]|))*)(:(\d+))?(\/[A-Z0-9~](([A-Z0-9_~-]|\.)*[A-Z0-9~]|))*\/?(.*)?$/i).test(value);
+                return (/^((http|https|ftp):\/\/)(([A-Z0-9][A-Z0-9_-]{1,62}\.)+([A-Z]{2,63}(\.[A-Z]{2,63})?)$)(:(\d+))?\/?/i).test(value);
 
             },
             $.mage.__('Please enter a valid URL. Protocol is required (http://, https:// or ftp://).')
         ],
         "validate-clean-url": [
             function(value) {
-                return utils.isEmptyNoTrim(value) || /^(http|https|ftp):\/\/(([A-Z0-9][A-Z0-9_-]*)(\.[A-Z0-9][A-Z0-9_-]*)+.(com|org|net|dk|at|us|tv|info|uk|co.uk|biz|se)$)(:(\d+))?\/?/i.test(v) || /^(www)((\.[A-Z0-9][A-Z0-9_-]*)+.(com|org|net|dk|at|us|tv|info|uk|co.uk|biz|se)$)(:(\d+))?\/?/i.test(value);
+                return utils.isEmptyNoTrim(value) || /^((http|https|ftp):\/\/)?(([A-Z0-9][A-Z0-9_-]{1,62}\.)+([A-Z]{2,63}(\.[A-Z]{2,63})?)$)(:(\d+))?\/?/i.test(value);
```
Comment (redelschaap) [inline app/code/Magento/Ui/view/base/web/js/lib/validation/rules.js]: It didn't check for the path, query and fragment parts correctly, my bad!

Correct return value:

```
                 return utils.isEmptyNoTrim(value) || /^((http|https|ftp):\/\/)?(([A-Z0-9][A-Z0-9_-]{1,62}\.)+([A-Z]{2,63}(\.[A-Z]{2,63})?))(:(\d+))?(\/[^\?#\s]*)?(\?[^#\s]*)?(#[^\s]*)?$/i.test(value);
```

File: app/code/Magento/Ui/view/base/web/js/lib/validation/rules.js [hunk 2]
```diff
@@ -468,14 +468,14 @@ define([
                     return true;
                 }
                 value = (value || '').replace(/^\s+/, '').replace(/\s+$/, '');
-                return (/^(http|https|ftp):\/\/(([A-Z0-9]([A-Z0-9_-]*[A-Z0-9]|))(\.[A-Z0-9]([A-Z0-9_-]*[A-Z0-9]|))*)(:(\d+))?(\/[A-Z0-9~](([A-Z0-9_~-]|\.)*[A-Z0-9~]|))*\/?(.*)?$/i).test(value);
+                return (/^((http|https|ftp):\/\/)(([A-Z0-9][A-Z0-9_-]{1,62}\.)+([A-Z]{2,63}(\.[A-Z]{2,63})?)$)(:(\d+))?\/?/i).test(value);
```
Comment (redelschaap) [inline app/code/Magento/Ui/view/base/web/js/lib/validation/rules.js]: It didn't check for the path, query and fragment parts correctly, my bad!

Correct return value:

```
                 return /^((http|https|ftp):\/\/)(([A-Z0-9][A-Z0-9_-]{1,62}\.)+([A-Z]{2,63}(\.[A-Z]{2,63})?))(:(\d+))?(\/[^\?#\s]*)?(\?[^#\s]*)?(#[^\s]*)?$/i.test(value);
```

**Review by redelschaap — CHANGES_REQUESTED** [review]
It didn't check for the path, query and fragment parts correctly, my bad!

**Comment by vrann** [comment]
@klict please update the code with the latest develop branch

**Comment by okorshenko** [comment]
Source branch has been removed. Closing this PR.

**Pull request closed without merging**